kratos/processes/find_conditions_neighbours_process.h

```diff
@@ -101,92 +104,41 @@ class FindConditionsNeighboursProcess
-        }
-
-        //adding the neighbouring conditions to the condition
-        //loop over faces
-        if (mTDim==3)
-        {
-            for(ConditionsContainerType::iterator ic = rConds.begin(); ic!=rConds.end(); ic++)
-            {
-                //face nodes
-                Geometry<Node >& geom = (ic)->GetGeometry();
-                //vector of the 3 faces around the given face
-                (ic->GetValue(NEIGHBOUR_CONDITIONS)).resize(3);
-                GlobalPointersVector< Condition >& neighb_faces = ic->GetValue(NEIGHBOUR_CONDITIONS);
-                //neighb_face is the vector containing pointers to the three faces around ic
-                //neighb_face[0] = neighbour face over edge 1-2 of element ic;
-                //neighb_face[1] = neighbour face over edge 2-0 of element ic;
-                //neighb_face[2] = neighbour face over edge 0-1 of element ic;
-                neighb_faces(0) = CheckForNeighbourFaces(geom[1].Id(), geom[2].Id(), geom[1].GetValue(NEIGHBOUR_CONDITIONS), ic->Id());
-                neighb_faces(1) = CheckForNeighbourFaces(geom[2].Id(), geom[0].Id(), geom[2].GetValue(NEIGHBOUR_CONDITIONS), ic->Id());
-                neighb_faces(2) = CheckForNeighbourFaces(geom[0].Id(), geom[1].Id(), geom[0].GetValue(NEIGHBOUR_CONDITIONS), ic->Id());
-            }
-        }
-    }
+    /**
+     * @brief Clears the neighbor data for all nodes and conditions.
+     */
+    void ClearNeighbours();
 
-
-    void ClearNeighbours()
-    {
-        NodesContainerType& rNodes = mr_model_part.Nodes();
-        for(NodesContainerType::iterator in = rNodes.begin(); in!=rNodes.end(); in++)
-        {
-            GlobalPointersVector<Condition >& rC = in->GetValue(NEIGHBOUR_CONDITIONS);
-            rC.erase(rC.begin(),rC.end());
-        }
-        ConditionsContainerType& rConds = mr_model_part.Conditions();
-        for(ConditionsContainerType::iterator ic = rConds.begin(); ic!=rConds.end(); ic++)
-        {
-            GlobalPointersVector<Condition >& rC = ic->GetValue(NEIGHBOUR_CONDITIONS);
-            rC.erase(rC.begin(),rC.end());
-        }
-
-    }
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    const Parameters GetDefaultParameters() const override;
 
     ///@}
     ///@name Access
     ///@{
 
-
     ///@}
     ///@name Inquiry
     ///@{
 
+    /**
+    * @brief Retrieve the IDs of neighboring conditions for each condition.
+    * @details This function retrieves the IDs of neighboring conditions for each condition in the model part.
+    * The IDs are stored in an unordered map where the key is the ID of the condition and the value
+    * is a vector containing the IDs of its neighboring conditions.
+    * @return An unordered map containing the IDs of neighboring conditions for each condition.
+    */
+    std::unordered_map<IndexType, std::vector<IndexType>> RetrieveConditionsNeighbourConditionsIds();
```

Comment:
This extends the API of the process and makes it not conformant with the base class one. This is a not very convenient thing that many of the old processes do and that we should in my opinion avoid for new implementations.

Hence, rather than to here, this most probably belongs to an external util. Also note that, if I'm not wrong, moving this to an external util would make possible to implement the very same for nodes and elements by adding a template argument.

Reply:
I will explain the reason of this. I need something to be in pure pythonm but when exposing `NEIGHBOUR_ELEMENTS` and `NEIGHBOUR_CONDITIONS` Kratos does not work in python anymore (pybind error that after 1 day I decided I was unable to fix, so this trick woulb be Okay for me)

Reply:
Makes sense